test/sslapitest.c

```diff
@@ -2027,6 +2239,97 @@ static int test_tlsext_status_type(void)
 
     return testresult;
 }
+
+static int test_tlsext_status_type_multi(void)
+{
+    SSL_CTX *cctx = NULL, *sctx = NULL;
+    SSL *clientssl = NULL, *serverssl = NULL;
+    int testresult = 0;
+    char *leaf_chain = test_mk_file_path(certsdir, "leaf-chain.pem");
+    char *skey = test_mk_file_path(certsdir, "leaf.key");
+    char *leaf = test_mk_file_path(certsdir, "leaf.pem");
+    OSSL_LIB_CTX *tmpctx;
+
+    tmpctx = OSSL_LIB_CTX_set0_default(libctx);
+
+    if (!create_ssl_ctx_pair(libctx, TLS_server_method(), TLS_client_method(),
+                             TLS1_VERSION, 0,
+                             &sctx, &cctx, leaf, skey))
+        return 0;
+
+    if (SSL_CTX_use_certificate_chain_file(sctx, leaf_chain) <= 0)
+        return 0;
```

Comment:
Use `goto end`, to free at least all results of `test_mk_file_path()`.

Reply:
You missed two cases where `goto end` is much simpler - I added respective suggestions.